src/Containers/MultiPseudoStack.cpp

```diff
@@ -99,6 +99,11 @@ void LockFreeQueue::write(qli_t *Q) {
     //printf("w%d ", free_elms+1);
 }
 
+int8_t LockFreeQueue::available()
```

Comment:
I'd through in a caveat that if this datastructure is being used in a multi-reader configuration (I forget if it is in practice), then this function returning a non-zero value does not imply that it will succeed in reading those bytes. This case should likely be documented with a quick comment, though the code is a-ok as it stands.